src/evaluators/UpdateExpression.js

```diff
@@ -27,23 +29,30 @@ export default function (ast: BabelNodeUpdateExpression, strictCode: boolean, en
 
   // Let oldValue be ? ToNumber(? GetValue(expr)).
   let oldExpr = GetValue(realm, expr);
-  if ((oldExpr instanceof AbstractValue) && IsToNumberPure(realm, oldExpr) &&
-      (ast.operator === '++' || ast.operator === '--')) {
-    let op = ast.operator === '++' ? '+' : '-';
-    let newAbstractValue = realm.createAbstract(
-      new TypesDomain(NumberValue),
-      ValuesDomain.topVal,
-      [oldExpr],
-      ([node]) => t.binaryExpression(op, node, t.numericLiteral(1))
-    );
-    PutValue(realm, expr, newAbstractValue);
-    if (ast.prefix) {
-      return newAbstractValue;
-    } else {
-      return oldExpr;
+  if (oldExpr instanceof AbstractValue) {
+    if (!IsToNumberPure(realm, oldExpr)) {
+      let error = new CompilerDiagnostics(
+        "might be a symbol or an object with an unknown valueOf or toString or Symbol.toPrimitive method",
+        ast.argument.loc, 'PP0008', 'RecoverableError');
+      if (realm.handleError(error) === 'Fail') throw fatalError;
+    }
+    if (ast.operator === '++' || ast.operator === '--') {
+      let op = ast.operator === '++' ? '+' : '-';
+      let newAbstractValue = realm.createAbstract(
+        new TypesDomain(NumberValue),
+        ValuesDomain.topVal,
+        [oldExpr],
+        ([node]) => t.binaryExpression(op, node, t.numericLiteral(1))
+      );
+      PutValue(realm, expr, newAbstractValue);
+      if (ast.prefix) {
+        return newAbstractValue;
+      } else {
+        return oldExpr;
+      }
     }
   }
-  oldExpr.throwIfNotConcrete();
+  invariant(oldExpr instanceof ConcreteValue);
```

Comment:
There are no update operators other than ++ and --, but good point, the if statement in line 39 is redundant in that case. I was seduced by the old code.